### ISSUE TITLE
Javascript Error occured in the main process. Failed to serialize arguments

### ISSUE THREAD

**Issue by Serge777if** [ytmdesktop/ytmdesktop#741]
- [v] I understand that **YTMDesktop have NO affiliation with Google or YouTube**.
- [v] I verified that there is no open issue for the same subject.

**Describe the bug**
Sometimes while playing my favorite songs playlist I got such error, I can close it, and app continue to work, but error may appear again.

**To Reproduce**
1. Start playing "favorite songs playlist"
2. Wait for some time (It appears randomly).

**Expected behavior**
no error

**Screenshots**
![YTM-error2](https://user-images.githubusercontent.com/71792530/121801671-2de73600-cc41-11eb-99c1-b9991caebc63.png)

**Environment (please complete the following information):**
 * YTMDesktop version: ***v1.13.0***
 * OS: ***Linux***
 * OS version: ***Ubuntu 20.04.1 LTS (Focal Fossa)***
 * Arch: ***x64***
* Installation way: ***AppImage***

**Comment by cwhits** [comment]
do you have notifications enabled? it looks to be related to that. I get the same message when I enable notifications on the mac client

**Comment by RubenSibon** [comment]
> do you have notifications enabled? it looks to be related to that. I get the same message when I enable notifications on the mac client

Yes, it could be related to notifications. After disabling them I have not seen the error again. But with notifications enabled it did not happen with every song, but only occasionally. I've seen this exact error message with the macOS as well as the Linux client.

**Comment by cwhits** [comment]
The notifications appear to only happen when the app is in the background. maybe that's why you only see it occasionally?

**Comment by RubenSibon** [comment]
Well, maybe, I tested it with the below procedure, but it seems to only happen the first time that a song changes while the app is in the background.

What I did on macOS: shut down the process; start the app; ensure that notifications are enabled; played a song and moved the scrobbler to the end of a song; closed the window; on the next song "Failed to serialize arguments" appeared.

**Comment by RubenSibon** [comment]
After testing some more: the error dialog also appears on first song change when the app's window is open. It seems that the error dialog only appears once per session.

**Comment by mingjun97** [comment]
Does this problem occurred at development branch? I guess someone had fixed this if I remember correctly.

**Comment by Alipoodle** [comment]
The application will not show notifications for when a new Song starts while the Windows is focused.
So starting a playlist or something initially won't show you the notification as the window has focus.
[Main.js Lines 488-493](https://github.com/ytmdesktop/ytmdesktop/blob/master/main.js#L488-L493)
```js
if (!mainWindow.isFocused() && settingsProvider.get('settings-show-notifications')) {
    tray.balloon(title, author, cover, iconDefault)
}
```

**Comment by RubenSibon** [comment]
> Does this problem occurred at development branch? I guess someone had fixed this if I remember correctly.

Yes, indeed, according to [this fix](https://github.com/ytmdesktop/ytmdesktop/commit/5fad6bf466313101c0fb2389973cac41d5670998) it should be fixed in development. These changes are not yet in master.

The issue is related to a dependency on electron-native-notifications: https://github.com/ytmdesktop/ytmdesktop/blob/b6b86526bb128915371659df82a6e869e9d46add/src/providers/trayProvider.js#L157

It does seem that this issue keeps coming back: #367, #516 so getting this into production soon would be good I suppose.

**Comment by Alipoodle** [comment]
Development != Master

Development is work for next app, and ironically the fix PR was made the week after 1.13 was made. 😅
But there's been a bit of talk recently within the Discord regarding making 1.14 release soon to fix a number of the issues & add a few features.

**Comment by Serge777if** [comment]
Thank you guys! After disabling notifications, seems like error is gone.
So, waiting for new release. )
